### PR TITLE
feature/SESC-7369_update_coding_challenge

### PR DESCRIPTION
**Pull request by jean-pauledoh**
Issue: https://gls-ce.atlassian.net/browse/SESC-7369

Updated the coding challenge to kotlin, added a few small mistakes and security config.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("org.springframework.boot") version "2.7.8"
+    id("io.spring.dependency-management") version "1.0.14.RELEASE"
+    kotlin("plugin.lombok") version "1.9.10"
```
Comment (kimakunc) [inline build.gradle.kts]: not need IMO

File: src/main/kotlin/com/gls/parcelshop/model/Address.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.gls.parcelshop.model
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import lombok.AllArgsConstructor
```
Comment (kimakunc) [inline src/main/kotlin/com/gls/parcelshop/model/Address.kt]: Hier ist noch lombok :)

**Review by kimakunc — CHANGES_REQUESTED** [review]

File: build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("org.springframework.boot") version "2.7.8"
+    id("io.spring.dependency-management") version "1.0.14.RELEASE"
+    java
```
Comment (valoricDe) [inline build.gradle.kts]: is this still needed?

Comment (valoricDe) [reply build.gradle.kts]: We also have kotlin("plugin.spring") version kotlinVersion no idea if this is necessary xD

**Review by valoricDe — APPROVED** [review]

**Review by valoricDe — APPROVED** [review]

**Comment by jean-pauledoh** [comment]
Closing this PR. The merged kotlin version is available here: https://github.com/bettermile/parcelshopK

**Pull request closed without merging**